### PR TITLE
Add additional tables to Memory view for drilling down

### PR DESCRIPTION
**Pull request by DanTup**
@devoncarew This adds multiple tables to the Memory page that appear/disappear based on selection (hopefully this is what you were describing - see below).

Currently it has hard-coded data because I'm not sure how to get the data. I did wire it up to `_getInstances` in the VM then realised that it was supposed to be a snapshot (most things returned `Sentinel` fetching them live!).

I can see there's a `_loadHeapSnapshot` function but it's commented out, and it only seems to result in a byte array. Is there any info on how this works I can look at? And do we need to be concerned about it being fetched separately to `_getAllocationProfile` (su the results might not match perfectly)?

screen shot 2018-09-19 at 3 02 04 pm
screen shot 2018-09-19 at 3 02 10 pm
screen shot 2018-09-19 at 3 02 15 pm

### CODE REVIEW

**Comment by devoncarew** [comment]
Yup, the UI looks great - let me look through the code.

**Comment by DanTup** [comment]
Ok, I changed the tables over to a stack so we can push new tables based on the selected data:

devtools

I also added a spinner for async work, though it's possible it doesn't make sense anywhere except waiting for the snapshot (there may not be much async work elsewhere, and things like getObject are very fast).

I might look at the data stuff on another branch that targets this one to simplify the PRs/changesets a little.

File: lib/memory/memory.dart [hunk 1]
```diff
@@ -79,6 +86,24 @@ class MemoryScreen extends Screen {
...
+    }
+
+    // Push the new table on to the stack and to the right of current.
+    if (next != null) {
+      tableStack.addLast(next);
+      tableContainer.add(next.element..clazz('margin-left'));
+      tableContainer.element.scrollTo(<String, dynamic>{
+        'left': tableContainer.element.scrollWidth,
+        'top': 0,
+        'behavior': 'smooth'
```
Comment (devoncarew) [inline lib/memory/memory.dart]: nit: add a trailing comma?

File: lib/memory/memory.dart [hunk 2]
```diff
@@ -102,8 +130,9 @@ class MemoryScreen extends Screen {
         return stats.instancesCurrent > 0; //|| stats.instancesAccumulated > 0;
       }).toList();
 
-      memoryTable.setRows(heapStats);
+      tableStack.first..setRows(heapStats);
```
Comment (devoncarew) [inline lib/memory/memory.dart]: too many periods (`first..setRows()`)

File: lib/memory/memory.dart [hunk 3]
```diff
@@ -174,40 +203,99 @@ class MemoryScreen extends Screen {
...
+        final Table<InstanceData> newTable =
+            row == null ? null : _createInstanceDetailsTableView(row.instance);
+        _pushNextTable(table, newTable);
+      } else {
+        _pushNextTable(table, null);
+      }
+    });
+
+    // Kick off population of data for the table.
+    // TODO: If it turns out not to be async work, remove the spinner.
```
Comment (devoncarew) [inline lib/memory/memory.dart]: TODO(dantup)

File: lib/memory/memory.dart [hunk 4]
```diff
@@ -25,7 +26,10 @@ class MemoryScreen extends Screen {
   StatusItem objectCountStatus;
 
   PButton loadSnapshotButton;
-  Table<ClassHeapStats> memoryTable;
+
+  // TODO(dantup): Is it reasonable to put dynamic here?
```
Comment (devoncarew) [inline lib/memory/memory.dart]: `Object` may be more correct here - the common super class of all the things we could display.

**Comment by devoncarew** [comment]
> LMK if you're happy with this being merged while it still has fake data.

Yes, sounds good. I think we'd still in the prototyping stage.

**Pull request merged**